temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -258,6 +321,32 @@ private RuntimeException createEventProcessingException(RuntimeException e, Hist
     }
   }
 
+  private void handleSingleMessage(Message message) {
+    // Get or create protocol state machine based on Instance ID and protocolName
+    EntityStateMachine stateMachine =
+        protocolStateMachines.computeIfAbsent(
+            message.getProtocolInstanceId(),
```

Comment:
The state machines are per protocol instance ID so that is what the map is, maybe I don't understand your question

Reply:
What I mean is (and maybe `protocol_instance_id` isn't what I think it is) the normal state machine map maps event IDs to state machines. If `protocol_instance_id` is something like `UPDATE_PROTOCOL` or whatever, how can we have two updates running at the same time? Seemingly they ought to be differentiated by their `id` rather than `protocol_instance_id`.

I'm sure I'm just missing something here, would be good to just clear it up, maybe that means changing the doc comment here https://github.com/temporalio/api/blob/e40424b5e4dd109fb3278e76820dc7990d009486/temporal/api/protocol/v1/message.proto#L44 since it sounds like that field isn't what I thought it was

Reply:
`protocol_instance_id ` is not `UPDATE_PROTOCOL ` . From the docs it says `Identifies the specific instance of a protocol` it says it identifies an instance of a protocol, not the protocol type itself

Reply:
Elsewhere in this PR you can see ProtocolUtils and ProtocolType which reflect the concepts you're thinking of, Spencer. Specifically, the ProtocolType is "temporal.api.update.v1" for this version of the protocol. Then we have _instances_ of the protocol. In practice these use the initially requested UpdateID to identify the collection of messages and state for a single update.  If you're running 10 updates then each will be identifiable by a protocolInstanceID